merlion/models/automl/autoets.py

```diff
@@ -30,3 +92,166 @@ class AutoETS(SeasonalityLayer):
     """
 
     config_class = AutoETSConfig
+
+    def generate_theta(self, train_data: TimeSeries) -> Iterator:
+        """
+        generate [theta]. theta is a list of parameter combination [error, trend, damped_trend, seasonal]
+        """
+        y = train_data.univariates[self.target_name].np_values
+
+        # check the size of y
+        n_samples = y.shape[0]
+        if n_samples <= 3:
+            self.information_criterion = "aic"
+
+        # auto-detect seasonality if desired, otherwise just get it from seasonal order
+        if self.config.auto_seasonality:
+            candidate_m = super().generate_theta(train_data=train_data)
+            self.config.seasonal_periods, _, _ = super().evaluate_theta(thetas=candidate_m, train_data=train_data)
+
+
+        # set the parameters ranges for error, trend, damped_trend and seasonal
+        if np.any(y <= 0):
+            E_range = ["add"]
+            T_range = ["add", None]
+        else:
+            E_range = ["add", "mul"]
+            if self.config.allow_multiplicative_trend:
+                T_range = ["add", "mul", None]
+            else:
+                T_range = ["add", None]
+
+        if self.config.seasonal_periods <= 1 or self.config.seasonal_periods is None or y.shape[0] <= self.config.seasonal_periods:
+            self.config.seasonal_periods = 1
+            S_range = [None]
+        elif np.any(y <= 0):
+            S_range = ["add", None]
+        else:
+            S_range = ["add", "mul", None]
+        D_range = [True, False]
+
+        if not self.config.auto_error:
+            E_range = [self.model.config.error]
+        if not self.config.auto_trend:
+            T_range = [self.model.config.trend]
+        if not self.config.auto_seasonal:
+            S_range = [self.model.config.seasonal]
+        if not self.config.auto_damped:
+            D_range = [self.model.config.damped_trend]
+
+        thetas = []
+        for error, trend, seasonal, damped in product(E_range, T_range, S_range, D_range):
+            if trend is None and damped:
+                continue
+            # if (error, trend, seasonal, damped) not in variants_pool_pred_interval:
+            #     continue
+            if self.config.additive_only:
+                if error == "mul" or trend == "mul" or seasonal == "mul":
+                    continue
+            if self.config.restrict:
+                if error == "add" and (trend == "mul" or seasonal == "mul"):
+                    continue
+                if error == "mul" and trend == "mul" and seasonal == "add":
+                    continue
+
+            thetas.append([error, trend, seasonal, damped])
+        return iter(thetas)
+
+    def evaluate_theta(
+        self, thetas: Iterator, train_data: TimeSeries, train_config=None, **kwargs
+    ) -> Tuple[Any, Optional[ETS], Optional[Tuple[TimeSeries, Optional[TimeSeries]]]]:
+        y = train_data.to_pd()[self.target_name]
```

Comment:
You should probably start with
```
model = deepcopy(self.model)
model.reset()
y = model.train_pre_process(train_data).to_pd()[self.target_name]
```
This is because the statsmodels ETS model will need to be trained on the pre-processed training data. Then, you can simplify the code below this loop to
```
self.set_theta(model, best_model_theta, train_data)
model.model = self._bestfit
times = y.index
```

Reply:
Thanks, I have updated this part. Besides, there exist some error for `plot_forecast` in the the tutorial of autosarima. It shows that 'AutoSarimaConfig' object has no attribute 'invert_transform'.

Reply:
I didn't get a chance to reproduce this error, but I think it should be solved after my most recent commits to this branch. I can successfully run the entire autosarima notebook.